File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dict1 = {}
    for p in participant:
        if(p in dict1): dict1[p] -= 1
        else: dict1[p] = 0
    for c in completion:
        dict1[c] += 1
    answer = ""
    for key, value in dict1.items():
        if value <= 0:
            answer = key
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

unordered_map<string, int> m;

string solution(vector<string> participant, vector<string> completion) {
    for(auto p : participant) m[p]++;
    for(auto c : completion) m[c]--;
    
    string answer = "";
    for(auto iter=m.begin(); iter!=m.end(); iter++) {
        if (iter->second > 0) answer += iter->first;
    }
    return answer;
}